cylc/flow/task_job_mgr.py

```diff
@@ -277,21 +277,58 @@ def submit_task_jobs(self, workflow, itasks, curve_auth,
-                    # Set the task state to submit-failed.
-                    itask.waiting_on_job_prep = False
-                    itask.local_job_file_path = None
-                    self._prep_submit_task_job_error(
-                        workflow, itask, '(remote init)', ''
-                    )
-                    self.bad_hosts.difference_update(itask.platform['hosts'])
-                    LOG.critical(TaskRemoteMgmtError(
+                    # See if you can get another platform from the group or
+                    # else set task to submit failed.
+                    LOG.warning(TaskRemoteMgmtError(
                         (
-                            'Initialisation on platform did not complete:'
-                            'no hosts were reachable.'
+                            'Tried all the hosts on platform.'
                         ), itask.platform['name'], [], 1, '', '',
                     ))
-                    out_of_hosts = True
-                    done_tasks.append(itask)
+                    # Get another platform, if task config platform is a group
+                    use_next_platform_in_group = False
+                    try:
+                        platform = get_platform(
+                            itask.tdef.rtconfig['platform'],
+                            bad_hosts=self.bad_hosts
+                        )
+                        # If were able to select a new platform;
+                        if platform and platform != itask.platform:
+                            use_next_platform_in_group = True
+                    except NoPlatformsError:
+                        use_next_platform_in_group = False
+
+                    if use_next_platform_in_group:
+                        # store the previous platform's hosts so that when
+                        # we record a submit fail we can clear all hosts
+                        # from all platforms from bad_hosts.
+                        for host_ in itask.platform['hosts']:
+                            self.bad_hosts_to_clear.add(host_)
+                        itask.platform = platform
+                        out_of_hosts = False
+                        break
+                    else:
+                        itask.waiting_on_job_prep = False
+                        itask.local_job_file_path = None
+                        self._prep_submit_task_job_error(
+                            workflow, itask, '(remote init)', ''
+                        )
+                        # Now that all hosts on all platforms in platform
+                        # group selected in task config are exhausted we clear
+                        # bad_hosts or all the hosts we have
+                        # tried for this platform or group.
+                        self.bad_hosts = (
+                            self.bad_hosts - set(itask.platform['hosts']))
+                        self.bad_hosts = (
```

Comment:
Manual check shows that this line _is_ touched by `/tests/functional/intelligent_host_selection/06...`
(boring)